src/core/hle/service/hid/controllers/touchscreen.h

```diff
@@ -30,6 +30,18 @@ class Controller_Touchscreen final : public ControllerBase {
     void OnLoadInputDevices() override;
 
 private:
+    static constexpr size_t MAX_FINGERS = 16;
+
+    // Returns an unused finger id, if there is no fingers avaliable MAX_FINGERS will be returned
+    size_t GetUnusedFingerID() const;
```

Comment:
```suggestion
    std::optional<size_t> GetUnusedFingerID() const;
```

And return `std::nullopt` to signal unavailability, rather than using an error-code value.